src/components/Chart/adaptDomainToData.ts

```diff
@@ -2,36 +2,44 @@ import { dateRangeStepToMillis } from './dateRangeStepToMillis'
 import { AxisDomain, ChartSeries, DateRange, isValueRange, ValueRange } from './model'
 import { getDataPointValue } from './util'
 
-export function adaptDomainToSeriesRange<XDomain>(domain: AxisDomain, series: ChartSeries<XDomain>[]): AxisDomain {
+export function adaptDomainToSeriesRange<XDomain>(
+  domain: AxisDomain,
+  series: ChartSeries<XDomain>[],
+  hasOutlier?: boolean
+): AxisDomain {
   const dataValues = series.flatMap((s) => (s.data as []).map((d) => getDataPointValue(d)))
-  return adaptDomainToDataRange(domain, dataValues)
+  return adaptDomainToDataRange(domain, dataValues, hasOutlier)
 }
 
-export function adaptDomainToDataRange(domain: AxisDomain, dataValues: number[] | Date[] | string[]): AxisDomain {
+export function adaptDomainToDataRange(
+  domain: AxisDomain,
+  dataValues: number[] | Date[] | string[],
+  hasOutlier?: boolean
+): AxisDomain {
   if (!domain || !dataValues) return domain
   if (Array.isArray(domain)) return dataValues as string[]
 
   const numericDataValues = (dataValues as []).map((d) => +d).filter((d) => !isNaN(d))
   if (!numericDataValues.length) return domain
 
-  if (isValueRange(domain)) return adaptValueRangeDomainToData(domain, numericDataValues)
-  else return adaptDateRangeDomainToData(domain, numericDataValues)
+  if (isValueRange(domain)) return adaptValueRangeDomainToData(domain, numericDataValues, hasOutlier)
+  else return adaptDateRangeDomainToData(domain, numericDataValues, hasOutlier)
 }
 
-function adaptValueRangeDomainToData(domain: ValueRange, dataValues: number[]): ValueRange {
+function adaptValueRangeDomainToData(domain: ValueRange, dataValues: number[], hasOutlier: boolean): ValueRange {
   if (domain.init >= domain.end) throw Error('Domain init must be less than domain end')
 
   const dataMin = Math.min(...dataValues)
   const dataMax = Math.max(...dataValues)
 
   return {
     init: dataMin < domain.init ? dataMin - (dataMin % domain.step) : domain.init,
-    end: dataMax > domain.end ? dataMax + (domain.step - (dataMax % domain.step)) : domain.end,
+    end: dataMax > domain.end && !hasOutlier ? dataMax + (domain.step - (dataMax % domain.step)) : domain.end,
```

Comment:
Hmmm, for me, hasOutliers seems more appropriate, because in this context, it's more about indicating the presence of outliers than displaying them. The component adapts the data based on the existence of outliers, what u think?